orttraining/tools/onnx2tfevents/onnx2tfevents.py

```diff
@@ -0,0 +1,379 @@
+                    "shape": AttrValue(shape=shape_proto),
+                    "parameters": AttrValue(s=attr),
+                },
+            )
+        )
+
+    return GraphDef(node=nodes, versions=VersionDef(producer=22))
+
+
+def add_onnx_model(self, model):
+    self._get_file_writer().add_onnx_graph(parse(model.graph))
+
+
+# Monkey patch SummaryWriter to add onnx graph.
+SummaryWriter.add_onnx_model = add_onnx_model
+
+
+class TransformerBase(ABC):
+    """Abstract base class for transformers that need to be applied to the ONNX graph before
+    converting it to TensorBoard. This base class is also used as a registry for all non-abstract transformers.
+    NOTE: Transformers are applied in the order they are registered.
+    """
+
+    _TRANSFORMERS = []
+
+    @classmethod
+    def register_transformer(cls, klass):
+        if not inspect.isabstract(klass):
+            cls._TRANSFORMERS.append(klass())
+
+    @classmethod
+    def run_transformers(cls, graph):
+        for transformer in cls._TRANSFORMERS:
+            transformer.transform(graph)
+
+    def __init_subclass__(cls):
+        super().__init_subclass__()
+        TransformerBase.register_transformer(cls)
+
+    @abstractmethod
+    def transform(self, graph):
+        pass
+
+    def _apply_to_all_names(self, graph, func):
+        for node in graph.node:
+            for idx, name in enumerate(node.input):
+                node.input[idx] = func(name)
+            for idx, name in enumerate(node.output):
+                node.output[idx] = func(name)
+        for node in itertools.chain(graph.input, graph.output, graph.initializer, graph.value_info):
+            node.name = func(node.name)
+
+
+class HierarchicalNameTransformer(TransformerBase):
+    """TensorBoard can show the graph in different level for better visualization. PyTorch exporter can already
+    add hierarchical information to the ONNX graph nodes, but the behavior is not consistent. For example,
+    for graph inputs and outputs, the exported name is _original_module.A.B.C, but for activations and initializers,
+    the exported name is /_original_module/A/B/C. This transformer will make the exported names consistent
+    to use "/" as the separator. The top level _original_module will also be removed.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.sections = set()
+        self.original_module_name = "_original_module"
+
+    def _add_sections(self, name):
+        if "/" in name:
+            for sec in name.split("/"):
+                if len(sec) > 0:
+                    self.sections.add(sec)
+
+    def _transform_name(self, name):
+        if "/" in name:
+            if name.startswith("/" + self.original_module_name + "/"):
+                name = name[len(self.original_module_name) + 2 :]
+            if name.startswith("/"):
+                name = name[1:]
+            return name
+        new_name = ""
+        while True:
```

Comment:
Possible to remove the while True loop? I would consider utilities in itertools if needed